### PR TITLE
Release 3 0 7

### PR DESCRIPTION
**Pull request by jacksonsj**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

## Summary of Proposed Changes
- Adds many analytics validation checks
- Cleans up code base to remove duplicate names and deprecate archived functions.

## Affected Process Elements
- [x] Data Pack generation
- [ ] Data Pack model
- [ ] Data Pack self-service app
- [x] Data Pack processing
- [ ] Site Tool model/distribution
- [ ] Site Tool generation (from Data Pack)
- [ ] Site Tool generation (from DATIM - for OPUs)
- [ ] Site Tool self-service app
- [ ] Site Tool to Data Pack comparison self-service app
- [ ] Site Tool to DATIM comparison self-service app
- [ ] Site Tool processing
- [ ] Site Tool import

## Types of changes

- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [x] Build-related changes
- [x] Documentation content changes
- [ ] Other (please describe): 

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

## Final Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: R/checkAnalytics.R [hunk 1]
```diff
@@ -1,44 +1,573 @@
...
+#' @title Check Data Pack data for retention < 98\% or >100\%
+#'
+#' @description Check data gathered from Data Pack to identify cases where
+#' retention is less than the standard of 98\% or >100\%.
+#'
+#' @param data Analytics object to analyze
+#'
+#' @return a
+#'
+analyze_retention <- function(data) {
```
Comment (sam-bao) [inline R/checkAnalytics.R:222]: @jacksonsj 

I am getting the following error when calling checkAnalytics. I am using this data pack: https://www.pepfar.net/Project-Pages/collab-38/Shared%20Documents/COP%2020%20Data%20Pack%20Staging%20Area/Completed%20Data%20Packs/Data%20Pack_Rwanda_20200403_v30.xlsx

unpacked first with unPackTool.


```
d <- datapackr::checkAnalytics(d, model_data_path = model_data_path)
Error in TX_CURR.N.Age_Sex_HIVStatus.T_1 + TX_NEW.N.Age_Sex_HIVStatus.T : 
  non-numeric argument to binary operator
In addition: Warning message:
Values are not uniquely identified; output will contain list-cols.
* Use `values_fn = list` to suppress this warning.
* Use `values_fn = length` to identify where the duplicates arise
* Use `values_fn = {summary_fun}` to summarise duplicates 
> traceback()
13: mutate_impl(.data, dots, caller_env())
12: mutate.tbl_df(., TX.Retention.T = (TX_CURR.N.Age_Sex_HIVStatus.T)/(TX_CURR.N.Age_Sex_HIVStatus.T_1 + 
        TX_NEW.N.Age_Sex_HIVStatus.T))
11: dplyr::mutate(., TX.Retention.T = (TX_CURR.N.Age_Sex_HIVStatus.T)/(TX_CURR.N.Age_Sex_HIVStatus.T_1 + 
        TX_NEW.N.Age_Sex_HIVStatus.T))
10: function_list[[k]](value)
9: withVisible(function_list[[k]](value))
8: freduce(value, `_function_list`)
7: `_fseq`(`_lhs`)
6: eval(quote(`_fseq`(`_lhs`)), env, env)
5: eval(quote(`_fseq`(`_lhs`)), env, env)
4: withVisible(eval(quote(`_fseq`(`_lhs`)), env, env))
3: data %<>% dplyr::mutate(TX.Retention.T = (TX_CURR.N.Age_Sex_HIVStatus.T)/(TX_CURR.N.Age_Sex_HIVStatus.T_1 + 
       TX_NEW.N.Age_Sex_HIVStatus.T))
2: analyze_retention(data)
1: datapackr::checkAnalytics(d, model_data_path = model_data_path)
```

**Comment by sam-bao** [comment]
Verified the following output is identical (except for some time stamps) under master and 3.0.7 for rwanda

- initial data pack build (compared excel files)
- initial PSNUxIM tab (compared excel files)
- repack psnuxIM tab (compared excels)
- identical data pack to datim comparisons (R df) when parsing a final data pack
- identical content (R df) when parsing a final data pack for : d$tests, d$data, and d$datim

Once the minor outstanding issues are fixed I will do the same comparison for South Africa.

File: R/checkAnalytics.R (hunk 1 above)
Comment (jacksonsj) [reply R/checkAnalytics.R:222]: @sam-bao - Try again using the code I just pushed here to this PR. Should address the issue you were seeing with the Rwanda Data Pack.

Summary of the issue:
`checkAnalytics` takes the DATIM import file as its starting point and then runs `adorn_import_file` to convert it to a denormalized dataset. One of the conversions this does is from DSD/TA to DP indicator_code. In doing this, I forgot to `summarize` to combine what would have been one line for DSD and another for TA into a single line representing our Data Pack indicator codes. This caused later functions to group these two values as list elements, which threw the error you were getting.

I also then found and addressed an issue in the function that checks for VMMC indeterminate issues caused by a simple misnaming of a data frame.

**Comment by jacksonsj** [comment]
@sam-bao - I'm now running into a build failure caused by `test_that("GetDataValueSets"), related to connecting with `play.dhis2.org/2.30`. The function tripping seems to be `datapackcommons::RetryAPI()`. Would you mind taking a look? The code is identical between here and what we've been using in `release-4-0-0`, so unclear what exactly is different. I tried repointing to `play.dhis2.org/2.33` but that didn't seem to solve much.

**Comment by sam-bao** [comment]
@jacksonsj I fixed the issue with the test and resolved a few conflicts. The checks on the branch and pull request now pass. I think I will still be blocked by this: https://github.com/pepfar-datim/datapackr/pull/179. I will need to have this merged into master and then pulled into this branch from master before I can finish testing.

**Pull request merged**